tests/integration/test_retry_middleware.py

```diff
@@ -0,0 +1,97 @@
+import pytest
+
+from msgraphcore.graph_session import GraphSession
+
+
+@pytest.fixture
+def graph_session():
+    scopes = ['user.read']
+    credential = _CustomTokenCredential()
+    session = GraphSession(credential, scopes)
+    return session
+
+
+class _CustomTokenCredential:
+    def get_token(self, scopes):
+        return ['{token:https://graph.microsoft.com/}']
+
+
+def test_no_retry_success_response(graph_session):
+    """
+    Test that a request with valid http header and a success response is not retried
+    """
+    response = graph_session.get(
```

Comment:
Any reasons why we are not using mock objects to test this instead of making actual http requests which may slow down the tests significantly?